trunk/SUAVE/Methods/Noise/Fidelity_One/Propeller/compute_harmonic_noise.py

```diff
@@ -0,0 +1,291 @@
+    k_x            = ((2*m*B*B_D*M_t)/(M_r*(1 - M_x*np.cos(theta_r))))      # wave number
+    k_y            = ((2*m*B*B_D)/(M_r*r))*((M_x - (M_r**2)*np.cos(theta_r))/(1 - M_x*np.cos(theta_r)))
+    phi_s          = ((2*m*B*M_t)/(M_r*(1 - M_x*np.cos(theta_r))))*(MCA/D)
+    S_r            = Y/(np.sin(theta_r))                                # distance in retarded reference frame   
+    Jmb            = jv(m*B,((m*B*r*M_t*np.sin(theta_r_prime))/(1 - M_x*np.cos(theta_r))))  
+    psi_L          = np.zeros((num_cpt,num_mic,num_prop,num_r,num_h))
+    psi_V          = np.zeros((num_cpt,num_mic,num_prop,num_r,num_h))
+
+    # normalized thickness  and loading shape functions                
+    psi_V[:,:,:,0,:]       = 2/3   
+    psi_L[:,:,:,0,:]       = 1     
+    psi_V[:,:,:,1:,:]      = (8/(k_x[:,:,:,1:,:]**2))*((2/k_x[:,:,:,1:,:])*np.sin(0.5*k_x[:,:,:,1:,:]) - np.cos(0.5*k_x[:,:,:,1:,:]))    
+    psi_L[:,:,:,1:,:]      = (2/k_x[:,:,:,1:,:])*np.sin(0.5*k_x[:,:,:,1:,:])                  
+
+    # sound pressure for thickness noise  
+    exponent_fraction = np.exp(1j*m_1d*B*((omega*S_r/a) +  phi_prime - np.pi/2))/(1 - M_x*np.cos(theta_r))
+    p_mT_H_function   = ((rho*(a**2)*B*np.sin(theta_r))/(4*np.sqrt(2)*np.pi*(Y/D)))* exponent_fraction
+    p_mT_H_integral   = np.trapz(((M_r**2)*(t_c)*np.exp(1j*phi_s)*Jmb*(k_x**2)*psi_V ),x = r[0,0,0,:,0], axis =3)
+    p_mT_H            = -p_mT_H_function[:,:,:,0]*p_mT_H_integral   
+    p_mT_H            = abs(p_mT_H)             
+
+    # sound pressure for loading noise 
+    p_mL_H_function   = (m_1d*B*M_t*np.sin(theta_r)/ (2*np.sqrt(2)*np.pi*Y*R_tip)) *exponent_fraction
+    p_mL_H_integral   = np.trapz((((np.cos(theta_r_prime)/(1 - M_x*np.cos(theta_r)))*dT_dr - (1/((r**2)*M_t*R_tip))*dQ_dr)
+                                  * np.exp(1j*phi_s)*Jmb * psi_L),x = r[0,0,0,:,0], axis = 3 )
+    p_mL_H            =  p_mL_H_function[:,:,:,0]*p_mL_H_integral 
+    p_mL_H            =  abs(p_mL_H)  
+
+    # unweighted harmonic sound pressure level 
+    #res.SPL_r[i,:,p_idx,:]      = 20*np.log10((np.linalg.norm(p_mL_H + p_mT_H, axis = 0))/p_ref) 
+    res.SPL_r        = 20*np.log10((abs(p_mL_H + p_mT_H))/p_ref) 
+    res.p_pref_r     = 10**(res.SPL_r/10)   
+    res.SPL_r_dBA    = A_weighting(res.SPL_r,res.f[:,:,:,0,:]) 
+    res.p_pref_r_dBA = 10**(res.SPL_r_dBA/10)         
+        
+    # convert to 1/3 octave spectrum 
+    res.SPL_prop_h_spectrum     = SPL_harmonic_to_third_octave(res.SPL_r,res.f[:,0,0,0,:],settings)         
+    res.SPL_prop_h_dBA_spectrum = SPL_harmonic_to_third_octave(res.SPL_r_dBA,res.f[:,0,0,0,:],settings)     
+  
+    # Rotational(periodic/tonal)   
+    res.SPL_prop_tonal_spectrum = 10*np.log10( 10**(res.SPL_prop_h_spectrum/10)) 
+ 
+    return  
+
+## @ingroupMethods-Noise-Fidelity_One-Propeller
+def vectorize_1(X,num_cpt, num_r ,num_prop,num_mic):
```

Comment:
you can combine all of these vectorize functions into one, then specify a flag for which type
(lines 157-291)

Reply:
Great Idea!! Changed